hazelcast/src/hazelcast/client/serialization.cpp

```diff
@@ -453,21 +457,33 @@ namespace hazelcast {
                 template<>
                 void data_output::write(int16_t value) {
                     if (is_no_write_) { return; }
-                    boost::endian::native_to_big_inplace(value);
+                    if (byte_order_ == boost::endian::order::big) {
+                        boost::endian::native_to_big_inplace(value);
+                    } else {
+                        boost::endian::native_to_little_inplace(value);
+                    }
                     output_stream_.insert(output_stream_.end(), (byte *) &value, (byte *) &value + util::Bits::SHORT_SIZE_IN_BYTES);
                 }
 
                 template<>
                 void data_output::write(int32_t value) {
                     if (is_no_write_) { return; }
-                    boost::endian::native_to_big_inplace(value);
+                    if (byte_order_ == boost::endian::order::big) {
```

Comment:
We should address little endianness on 
`write(char i) ` and `write(char16_t i) ` as well.